apps/theming-designer/src/components/Header.tsx

```diff
@@ -0,0 +1,157 @@
+const codepenHeader = 'const { loadTheme, DefaultButton, PrimaryButton, Toggle, TooltipHost } = Fabric;\n\n';
+const codepenSamples =
+  '\n\nclass Content extends React.Component {\n  public render() {\n    return (<div>' +
+  '<DefaultButton text="DefaultButton"/><PrimaryButton text="PrimaryButton"/>' +
+  '<Toggle label="Enabled"/><Toggle label="Disabled" disabled={true}/>' +
+  '</div>);\n  }\n}\n' +
+  "ReactDOM.render(<Content />,document.getElementById('content'));";
+
+let jsonTheme: string;
+let powershellTheme: string;
+let themeAsCode: any;
+
+export class Header extends React.Component<IHeaderProps, IHeaderState> {
+  constructor(props: any) {
+    super(props);
+    this.state = {
+      showPanel: false
+    };
+  }
+
+  public render(): JSX.Element {
+    return (
+      <Card styles={{ root: { width: '1200px', height: '25px' } }}>
+        <Stack horizontal gap={800}>
+          <Stack horizontal>
+            <IconButton
+              disabled={false}
+              checked={false}
+              iconProps={{ iconName: 'WindowsLogo', styles: { root: { fontSize: '20px' } } }}
+              ariaLabel="Microsoft UI Fabric | Theming Designer"
+            />
+            <Label>Microsoft UI Fabric | Theming Designer</Label>
+          </Stack>
+          <Stack horizontal>
+            <IconButton
+              data-automation-id="test"
+              disabled={false}
+              checked={false}
+              iconProps={{ iconName: 'Export', styles: { root: { fontSize: '20px' } } }}
+              ariaLabel="Export to JSON/Code/PowerShell"
+              onClick={this.showPanel}
+              allowDisabledFocus={true}
+            />
+            <Label>Export theme</Label>
+            <Panel
+              isOpen={this.state.showPanel}
+              type={PanelType.smallFixedFar}
+              onDismiss={this.hidePanel}
+              headerText="Export theme"
+              closeButtonAriaLabel="Close"
+              onRenderFooterContent={this.onRenderFooterContent}
+            >
+              <span>
+                <p>
+                  This code block initializes the theme you have configured above and loads it using the loadTheme utility function. Calling
+                  loadTheme will automatically apply the configured theming to any Fabric controls used within the same app. You can also
+                  export this example to CodePen with a few component examples below.
+                </p>
+              </span>
+              <div className={outputPanelClassName}>
+                <Pivot>
+                  <PivotItem headerText="Code">
+                    <textarea className={textAreaClassName} readOnly={true} spellCheck={false} value={codeHeader + themeAsCode} />
+                  </PivotItem>
+                  <PivotItem headerText="JSON">
+                    <textarea className={textAreaClassName} readOnly={true} spellCheck={false} value={jsonTheme} />
```

Comment:
Referencing comment

> Since these variables directly affect rendering they should probably be stored in state.

Could also make `value` equal to `JSON.stringify(ThemeGenerator.getThemeAsJson(abridgedTheme), void 0, 2)` insetad

Reply:
abridgedTheme is calculated separately in exportToJson so I can't use it here inside render without making that a state variable too - since I started storing jsonTheme in state I'll just go with jsonTheme here.